homeassistant/components/sensor/darksky.py

```diff
@@ -188,6 +203,17 @@ def unit_system(self):
         """Return the unit system of this entity."""
         return self.forecast_data.unit_system
 
+    @property
+    def entity_picture(self):
+        """Return the entity picture to use in the frontend, if any."""
+        if self._icon is None or "summary" not in self.type:
```

Comment:
Rechecked. I set `self._icon` to `None` in the constructor. I only set it to a value (or empty string, if `getattr` fails) for specific types of the sensor (namely, all types that have `summary` in their name). So the first part of the condition is correct - I'm looking at cases when `self._icon` hasn't been set since construction time. You could say that the second part of the condition is redundant - if you think it doesn't belong there, I'll drop it.